### DISCUSSION TITLE
Secure RA and adminweb

### DISCUSSION THREAD

**Discussion by MRLeflei** [Keyfactor/ejbca-ce#738] [General]
Hi, How do I change the certificate so that I do not get certificate warnings when accessing the UI? The managementCA is the issueing CA now, Can I for example use a certificate from another CA for secure the web interface?

**Comment by mike-agrenius-kushner** [comment]
Hi,

The way the browsers handle TLS is that the root certificate has to be from a known and trusted issuer – this is of course a story unto itself, and you'd have to go through the process of getting an internal-use certificate from one of the major commercial CAs. Unless you're a major enterprise yourself, I suspect that this isn't the path you're aiming for,

The much easier answer is that for your internal use you can simply add the ManagementCA certificate to your local root store, either locally on your machine (if using a webkit-derived browser like Chrome or Safari) or in the browser itself if using Firefox. 

Cheers,
Mike

**Reply by MRLeflei** [reply]
Hi Mike,

Thanks for your reply. I think is whould be nice if I could change the certifcate so we are consitent for all users. But just so I understand. I have setup up a address ejbca.company.com that can be reached internaly. When I use firefox to access that address I get an certificate error since ejbca.company.com does not match what is created when deploying ejbca. So Can I change the ejbca cert to reflect ejbca.company.com or is the best practice to just leave it as is and live with the cert error?

**Reply by mike-agrenius-kushner** [reply]
Could you screenshot your error, because it's not 100% making sense to me =)

For reference, the tls cert I have deployed locally in my dev environment doesn't have any dnsName SAN set at all – though I have set it as trusted in my laptops certificate store. 

<img width="541" alt="Screenshot 2024-12-11 at 13 54 23" src="https://github.com/user-attachments/assets/8744641e-c3e1-433d-a071-95303eb0016a">

**Reply by MRLeflei** [reply]
Hi Mike, I have installed EJBCA CE in kubernetes and plan to use it in production. So it is not installed locally. So when I from my laptop access https://ejbca.company.local I get an cert error. So I want to swap out the https certificate to something that is valid when accessing  https://ejbca.company.local.

**Reply by svenska-primekey** [reply]
Are you using ingress or terminating TLS at the EJBCA pod in Kubernetes? Did you use the community EJBCA helm chart to deploy EJBCA?

**Reply by MRLeflei** [reply]
Sorry, openshift not kubernets. And I have not used the community helm chart

**Reply by svenska-primekey** [reply]
I would recommend using the helm chart to deploy. The tutorial in this link has a step for setting up the nginx in the EJBCA pod, creates a secret for the TLS certificate you want to use, and then deploy EJBCA with helm.

https://docs.keyfactor.com/ejbca/latest/tutorial-clean-up-microk8s-cluster-and-redeploy#id-(9.1.1)Tutorial-CleanupMicroK8sClusterandRedeploywithHelm-Step6-DeployEJBCAwithHelm

**Reply by MRLeflei** [reply]
Jag kollar upp det! återkommer

**Reply by MRLeflei** [reply]
So I almost got it to work in openshift using a route and reencrypt, but now firefox won't let me use the superadmin certificate. Any advice?

**Reply by svenska-primekey** [reply]
It sounds like the route needs to be configured with the ca certificates to trust for mtls to prompt you for the super admin certificate. With nginx ingress the annotation `nginx.ingress.kubernetes.io/auth-tls-secret` is used. There should be something comparable for OpenShift. I'm not sure what that is as I have no experience using OpenShift yet.
